configFlannel.md

```diff
@@ -115,11 +119,15 @@ FLANNEL_OPTIONS="eth0"
 ```
 
 
-* Enable the flanneld service and reboot.
+* Enable the flanneld service, let systemd know that the local flanneld service needs a default route and the local etcd service to be started before it, and reboot.
 
 
 ```
-# systemctl enable flanneld
+# systemctl enable NetworkManager-wait-online.service
+# systemctl enable flanneld.service
+# mkdir /etc/systemd/system/flanneld.service.wants
+# ln -s /usr/lib/systemd/system/etcd.service /etc/systemd/system/flanneld.service.wants/
+# ln -s /usr/lib/systemd/system/NetworkManager-wait-online.service /etc/systemd/system/flanneld.service.wants/
```

Comment:
I think you may be right, that the startup problems I was seeing could have been caused by not being up yet.  This is what happened when etcd was being started later:

`Jun 23 14:06:55 nalin-atomic-1.os1.phx2.redhat.com flanneld[697]: E0623 14:06:36.579070 00697 main.go:151] Failed to create SubnetManager: 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]`
`Jun 23 14:06:55 nalin-atomic-1.os1.phx2.redhat.com flanneld[697]: E0623 14:06:37.581440 00697 main.go:151] Failed to create SubnetManager: 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]`
`Jun 23 14:06:55 nalin-atomic-1.os1.phx2.redhat.com flanneld[697]: E0623 14:06:38.582929 00697 main.go:151] Failed to create SubnetManager: 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]`
`Jun 23 14:06:55 nalin-atomic-1.os1.phx2.redhat.com flanneld[697]: I0623 14:06:41.377479 00697 main.go:247] Installing signal handlers`
`Jun 23 14:06:55 nalin-atomic-1.os1.phx2.redhat.com flanneld[697]: I0623 14:06:41.377529 00697 main.go:118] Determining IP address of default interface`
`Jun 23 14:06:55 nalin-atomic-1.os1.phx2.redhat.com flanneld[697]: E0623 14:06:41.378311 00697 main.go:190] Failed to get default interface: Unable to find default route`
`Jun 23 14:06:55 nalin-atomic-1.os1.phx2.redhat.com flanneld[697]: I0623 14:06:41.378335 00697 main.go:264] VXLAN mode exited`

They're still being started in the right order on my system, even after removing the .wants symlink, so I'm not seeing this problem any more.